lib/jarvis/defer.rb

```diff
@@ -0,0 +1,15 @@
+module Jarvis class Defer
+  def initialize
+    @deferred = []
+  end
+
+  def do(&block)
+    @deferred << block
+  end
+
+  def run
+    @deferred.each do |block|
+      block.call
+    end
+  end
+end end
```

Comment:
We could probably run the defer call with concurrent ruby future? I don't think the order of execution is really important in that context.